jujugui/static/gui/src/app/app.js

```diff
@@ -1157,6 +1157,33 @@ YUI.add('juju-gui', function(Y) {
       this.navigate(url);
     },
 
+    /** Chooses an env to connect to from the env list based on config.
+
+      @method _pickEnv
+      @param {String} envList The list of environments to pick from.
+      @return {Object} The selected environment.
+     */
+    _pickEnv: function(envList) {
+      // XXX This picks the first environment if one is not provided by
+      // config, but we'll want to default to sandbox mode then allow the
+      // user to choose an env if one isn't provided in config.
+      var envName = this.get('jujuEnvUUID'),
+          user = this.get('user'),
+          envData;
+      if (envName && user) {
+        var path = user + '/' + envName;
+        envData = envList.filter(function(env) {
+          return env.path === path;
+        });
+      }
+      if (envData.length !== 0) {
```

Comment:
This will fail with a syntax error in the event that the conditional on line 1173 evaluates to false because then `envData` will be undefined. If you don't want to change the logic of this method you'll want to `var envData = [];`